pyinteraph/reformat_interaction_network_graph.py

```diff
@@ -0,0 +1,97 @@
+import sys
+
+import argparse
+import functools
+import logging
+import networkx as nx
+import numpy as np
+import MDAnalysis as mda
+import warnings
+
+
+from pyinteraph.file_validation import ArgumentParserFileExtensionValidation
+
+warnings.filterwarnings("ignore")
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.INFO,
+    format="[%(asctime)s] %(levelname)s | %(message)s",
+    datefmt="%Y/%m/%d %H:%M:%S"
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ReformatDatGraph:
+    def __init__(self, interaction_network_file, output_name, reference_structure_file=None):
+        self.reference_structure_file = reference_structure_file
+        self.interaction_network_file = interaction_network_file
+        self.output_name = output_name
+
+    @property
+    @functools.lru_cache()
+    def interaction_network(self):
+        try:
+            interaction_network = np.loadtxt(self.interaction_network_file)
+            return interaction_network
+        except:
+            sys.exit(1)
+
+    @property
+    @functools.lru_cache()
+    def reference_structure(self):
+        if not self.reference_structure_file:
+            return
+        try:
+            return mda.Universe(self.reference_structure_file)
+        except:
+            logger.warning("Reference structure not readable.")
+            return
+
+    @property
+    @functools.lru_cache()
+    def node_names(self):
+        if not self.reference_structure:
+            logger.warning("Auto-generated numbers will be used to rename nodes.")
+            return [str(i) for i in range(1, self.interaction_network.shape[0] + 1)]
+        return list(map(lambda r: f"{r.segment.segid}-{r.resnum}{r.resname}", self.reference_structure.residues))
+
+    @property
+    def interaction_network_graph(self):
+        try:
+            uploaded_interaction_network_graph = nx.Graph(self.interaction_network)
+            node_names = dict(zip(range(self.interaction_network.shape[0]), self.node_names))
+            nx.relabel_nodes(uploaded_interaction_network_graph, mapping=node_names, copy=False)
+            return uploaded_interaction_network_graph
+        except:
+            logger.error("Reformatted interaction network graph not generated. "
+                         "Please check the content of the interaction network file")
+            sys.exit(1)
+
+    @property
```

Comment:
I don't think this should be a `@property` since we are not setting anything useful. We are just performing the conversion